src/plugins/region_map/public/components/map_choice_options.tsx

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './map_choice_options.scss';
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiCheckableCard,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiTextColor,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { FormattedMessage } from '@osd/i18n/react';
+import { VisOptionsProps } from 'src/plugins/vis_default_editor/public';
+import { FileLayerField, VectorLayer, IServiceSettings } from '../../../maps_legacy/public';
+import { SelectOption, SwitchOption } from '../../../charts/public';
+import { RegionMapVisParams } from '../../../maps_legacy/public';
+import { DEFAULT_MAP_CHOICE, CUSTOM_MAP_CHOICE } from '../../common';
+
+const mapLayerForOption = ({ layerId, name }: VectorLayer) => ({
+  text: name,
+  value: layerId,
+});
+
+const mapFieldForOption = ({ description, name }: FileLayerField) => ({
+  text: description,
+  value: name,
+});
+
+const mapCustomJoinFieldForOption = ({ description, name }: FileLayerField) => ({
+  label: name,
+  text: description,
+  value: name,
+});
+
+export type MapChoiceOptionsProps = {
+  getServiceSettings: () => Promise<IServiceSettings>;
+} & VisOptionsProps<RegionMapVisParams>;
+
+function MapChoiceOptions(props: MapChoiceOptionsProps) {
+  const { getServiceSettings, stateParams, vis, setValue } = props;
+  const vectorLayers = vis.type.editorConfig.collections.vectorLayers;
+  const customVectorLayers = vis.type.editorConfig.collections.customVectorLayers;
+  const vectorLayerOptions = useMemo(() => vectorLayers.map(mapLayerForOption), [vectorLayers]);
+  const customVectorLayerOptions = useMemo(() => customVectorLayers.map(mapLayerForOption), [
+    customVectorLayers,
+  ]);
+  const [userSelectedLayer, setUserSelectedLayer] = useState(DEFAULT_MAP_CHOICE);
+
+  const fieldOptions = useMemo(
+    () =>
+      ((stateParams.selectedLayer && stateParams.selectedLayer.fields) || []).map(
+        mapFieldForOption
+      ),
+    [stateParams.selectedLayer]
+  );
+
+  const customFieldOptions = useMemo(
+    () =>
+      ((stateParams.selectedCustomLayer && stateParams.selectedCustomLayer.fields) || []).map(
+        mapCustomJoinFieldForOption
+      ),
+    [stateParams.selectedCustomLayer]
+  );
+
+  const selectDefaultVectorMap = () => {
+    setUserSelectedLayer(DEFAULT_MAP_CHOICE);
+    setValue('layerChosenByUser', DEFAULT_MAP_CHOICE);
```

Comment:
Why do we need to duplicate the state? don't you already have access to it in `stateParams`?

Reply:
`layerChosenByUser` is set from the UI when user either selects 'default' or the 'custom' vector map option. By default, the selected option is 'default'.

Reply:
I got that, but why store that info in two locations? one in useState and another time in `stateParams`. `layerChosenByUser` is a part of `stateParams`. If you are setting the value there, you dont need to store it again in useState. unless im missing some other reason

Reply:
`layerChosenByUser` is getting added to `stateParams`  here based on the choice what user makes and then used further in choropleth layer. While we need the default state to show something when the tab loads, we make use of `userSelectedLayer` variable. Let me know if this can be simplified.

Reply:
Exactly. `stateParams` can also have default values, however they have to be defined here: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/region_map/public/region_map_type.js#L54

Reply:
Got it! Will add it here : ) thanks!